### DISCUSSION TITLE
Ads Setup status endpoint?

### DISCUSSION THREAD

**Discussion by ecgan** [woocommerce/google-listings-and-ads#450] [Q&A]
@mikkamp , I happened to see your PR https://github.com/woocommerce/google-listings-and-ads/pull/443 and I have a side-topic question: can we have a status endpoint for ads setup too? I'm currently looking into https://github.com/woocommerce/google-listings-and-ads/issues/446 and it is suggested to use `glaData.adsSetupComplete`, however if we have an endpoint to retrieve ads setup status (similar to your PR for MC setup status), I think that would be better than using `glaData`.

**Comment by mikkamp** [comment]
With the merchant center we have the account setup status, which is four steps: creation, verification, linking, claiming
The `mc/connection` endpoint tells us what step is incomplete.

There is also three separate onboarding steps: account, target_audience, shipping_taxes
The `mc/setup` endpoint tells us what step is incomplete.

For ads we have multiple account steps: create/link account, billing flow, link to merchant, conversion action
The `ads/connection` endpoint will tell us at what step is incomplete.

If we would also create `ads/setup`, what kind of information would we add there that we don't already have access to?

**Reply by ecgan** [reply]
Okay, I think I have a bit clearer picture now. 

> The `ads/connection` endpoint will tell us at what step is incomplete.

So when the ads onboarding setup is complete, the `ads/connection` endpoint would return us `status: "connected"`? That's what I am seeing. 

However, when I go to browser console and check for `glaData`, it says `glaData.adsSetupComplete` is false. So there is a discrepancy between the two? 🤔 

@eason9487 , I guess you might have more info on the `glaData.adsSetupComplete`? 

The `glaData.adsSetupComplete` is added in https://github.com/woocommerce/google-listings-and-ads/pull/346.

Ultimately, my main question is: can we use the `ads/connection` endpoint's response `status: "connected"` to determine ads setup is completed (and hence we don't need to have `glaData.adsSetupComplete`, and we don't need to do force reload in the browser to refresh `glaData`)?

**Reply by mikkamp** [reply]
The final steps were implemented here: #403
